### ISSUE TITLE
fail to fall back when sparse arrays are passed to MKLDNN-enabled operators.

### ISSUE THREAD

**Issue by zheng-da** [apache/incubator-mxnet#11448]
Currently, the MKLDNN-enabled operators, such as convolution and pooling, can't handle sparse arrays correctly. The reason is that the storage inference of these operators doesn't return the right dispatch mode.
The MKLDNN-enabled operators include:
- [ ] Activation @azai91 
- [x] BatchNorm @zheng-da
- [ ] Convolution @ZhennanQin  
- [ ] Deconvolution @ZhennanQin 
- [x] FullyConnected @haojin2 
- [ ] LRN @azai91 
- [ ] Pooling @luobao-intel
- [ ] Softmax @luobao-intel

We may also need to test the operators below:
- [x] Concat
- [ ] Copy
- [ ] Sum
- [ ] Flatten

@haojin2 @azai91 @pengzhao-intel @TaoLv

**Comment by pengzhao-intel** [comment]
@zheng-da Could you help provide a test case?  
Does cuDNN backend handle these cases?

**Comment by frankfliu** [comment]
Thank you for submitting the issue!  @sandeep-krishnamurthy requesting this be labeled.

**Comment by zheng-da** [comment]
@eric-haibin-lin  @haojin2  could you please provide a test case?

**Comment by eric-haibin-lin** [comment]
There’s an existing test batch norm training test disabled due to flakyness

**Comment by eric-haibin-lin** [comment]
concat is tested and fixed.

**Comment by haojin2** [comment]
I'll take FullyConnected if no one's working on it right now.

**Comment by zheng-da** [comment]
i created a PR for batchnorm.

**Comment by pengzhao-intel** [comment]
what else need to cover? @zouluobao can help

**Comment by pengzhao-intel** [comment]
@luobao-intel will handle other OPs :)

**Comment by zheng-da** [comment]
@pengzhao-intel @luobao-intel we need to fix them and merge the PRs to the v1.3 release, which is probably before the end of next week.

**Comment by haojin2** [comment]
@zheng-da fullyconnected is done in #11498, if you want to add the MKLDNN backward storage type inference for it please go ahead and do so. Thanks!

**Comment by zheng-da** [comment]
@haojin2 I thought you have fixed its backward storage type inference.

**Comment by haojin2** [comment]
@zheng-da checkout https://github.com/apache/incubator-mxnet/pull/11498/files#diff-f2984a387afd7a02b74fe43b9ef50ee1R214

**Comment by pengzhao-intel** [comment]
@zheng-da related PR are merged. Could you verify if all issues are fixed and close the issue?
